src/groupeddataframe/utils.jl

```diff
@@ -5,7 +5,9 @@ function hashrows_col!(h::Vector{UInt},
                        v::AbstractVector{T},
                        rp::Nothing,
                        firstcol::Bool) where T
-    @spawn_for_chunks 1_000_000 for i in eachindex(h)
+    min_chunk_size = clamp(length(h) ÷ Threads.nthreads(),
+                           100_000, 10_000_000)
+    @spawn_for_chunks min_chunk_size for i in eachindex(h)
```

Comment:
The idea with `@spawn_for_chunks` was that the code can be kept clean and simple, with computations being performed by the macro in a centralized place. How about moving this logic there?

There may be reasons to spawn more tasks than threads, since if some threads are busy doing other things, that will allow distributing work across remaining threads more evenly. Is that why you chose a max at 10M rows?

Reply:
We are dealing here with relatively cheap (but not super cheap) operations. Therefore I put the `min_chunk_size` definition here.

However, I benchmarked more. The cost of spawning one task is around 3μs (@tkf - is my estimate correct?). This means that if we set `100_000` as chunk size the overhead with 10^10 rows is around 0.03 seconds which will be negligible. Therefore I think it is OK to just change it to `100_000` to keep it simple.

The rationale is as follows:
```
julia> x = trues(10^10);

julia> @time sum(x)
  0.074198 seconds (1 allocation: 16 bytes)
10000000000

julia> @time sum(hash, x)
 11.021615 seconds (247.28 k allocations: 11.731 MiB, 0.62% compilation time)
0xcebfde26f6717800
```
So we see that if we had an extremely cheap operation (like addition) using `min_chunk_size` equal to `100_000` would be too slow, but in the code that we have here we do hashing, and then the overhead is negligible.

My previous proposal was aimed to cover all the cases (also the super-fast operations, where 10M threshold seemed safe), but probably this is not needed.